velox/expression/Expr.cpp

```diff
@@ -1839,6 +1900,24 @@ void ExprSet::eval(
     context.ensureFieldLoaded(field->index(context), rows);
   }
 
+  if (FLAGS_experimental_velox_save_input_on_fatal_signal) {
+    auto other = process::GetThreadDebugInfo();
+    process::ThreadDebugInfo debugInfo;
+    if (other) {
+      debugInfo.queryId_ = other->queryId_;
+      debugInfo.taskId_ = other->taskId_;
+    }
+    debugInfo.callback_ = [&]() {
+      printInputAndExprs(context.row(), this->exprs());
+    };
+    process::ScopedThreadDebugInfo scopedDebugInfo(debugInfo);
+
+    for (int32_t i = begin; i < end; ++i) {
```

Comment:
nit: Only reason we need to duplicate the for loop is because of the scopedcDebugInfo - is there a way to make that a no-op and not have the loop twice ?

Reply:
Yes thats exactly the reason. I could not find any other way. I thought of moving the loop in a utility method, but its small enough that its not worth moving it to one.